holidays/holiday_base.py

```diff
@@ -302,17 +306,16 @@ def __contains__(  # type: ignore[override]
           :func:`dateutil.parser.parse`,
         * or a :class:`float` or :class:`int` representing a POSIX timestamp.
         """
-        return dict.__contains__(self, self.__keytransform__(key))
+        if not isinstance(key, (date, datetime, str, float, int)):
+            raise TypeError("Cannot convert type '%s' to date." % type(key))
```

Comment:
IMO `return False` might be better here, but you have unit tests checking for the current behaviour

Reply:
Hi @dimbleby I understand / agree on the check on the date format -> exception, but why did you remove the return value (True / False) from the __contains__ definition? :-/

Reply:
Oh right, it was hidden by getitem code in fact, sorry!
Now I see your point: I definitely prefer the date format check as it is, ie: raise an exception in case of unknown format 👍